presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -239,6 +250,35 @@ private <C> void createQueryInternal(QueryId queryId, String slug, int retryCoun
         }
     }
 
+    private void checkPermissions(QueryId queryId, SessionContext sessionContext)
+    {
+        Identity identity = sessionContext.getIdentity();
+        // When selectAuthorizedIdentity API is not enabled, only check delegation permission
+        if (!securityConfig.isAuthorizedIdentitySelectionEnabled()) {
+            accessControl.checkCanSetUser(
+                    identity,
+                    new AccessControlContext(
+                            queryId,
+                            Optional.ofNullable(sessionContext.getClientInfo()),
+                            Optional.ofNullable(sessionContext.getSource())),
+                    identity.getPrincipal(),
+                    identity.getUser());
+        }
```

Comment:
There is no return in the if block, so the new code will always be executed, is that the intention ?

Reply:
Done.